benefit-finder/cypress/e2e/storybook/dataLayer.cy.js

```diff
@@ -217,11 +217,28 @@ describe('Calls to Google Analytics Object', function () {
         .button()
         .contains(EN_LOCALE_DATA.intro.button)
         .then(() => {
-          // get the last pushed event
-          const ev = { ...window.dataLayer[window.dataLayer.length - 1] }
-          removeID(ev)
+          cy.wait(wait).then(() => {
+            assert.isDefined(
+              window.dataLayer.find(x => x.event === 'gtm.load'),
+              'GTM is done loading'
+            )
+            assert.isDefined(
+              window.dataLayer.find(x => x.event === 'bf_page_change'),
+              'bf_page_change is loaded'
+            )
 
-          expect(ev).to.deep.equal(dataLayerValueIntro)
+            cy.wait(500).then(() => {
+              console.log(window.dataLayer)
+              // get the last pushed event
+              const bfEventIndex = window.dataLayer.findIndex(
```

Comment:
Sometimes resourcing was creating a situation where the `gtm.load` event was in the `dataLayer` after the `bf-*-<event>`. When we asserted against the last  indexed event in the `dataLayer` array, it was not our `bf-*-<event>`. All though it was in the array. Now, we find the event index and target that in our assertion.